tests/shared/src/test/scala/scala/meta/tests/parsers/PositionSuite.scala

```diff
@@ -62,4 +62,16 @@ class PositionSuite extends ParseSuite {
         | List(Term.Name{8..9}("c")))
     """.trim.stripMargin.split("\n").mkString)
   }
+
+  test("apply(startLine,startColumn,endLine,endColumn)") {
+    val input = Input.String(
+      """|  val x = 2 // line 0
+         |            // line 1""".stripMargin
+    )
+    val x = Position.Range(input, 0, 2, 0, 11)
+    assert(x.text == "val x = 2")
+    val last = Position.Range(input, 1, 0, 1, Int.MaxValue)
+    assert(last.text == "            // line 1")
+  }
```

Comment:
Also, when thinking how to test this, I noticed that even with this change, we're not quite replicating LSP's behavior. To quote the spec (https://microsoft.github.io/language-server-protocol/specification):

> Positions are line end character agnostic. So you can not specify a position that denotes \r|\n or \n| where | represents the character offset.

I don't think this can work for Scalameta. Up until now it was possible to construct positions like that, so I don't think we can go ahead and outright ban these positions.

This raises a new question. If we can't fully support LSP conventions regarding positions, would it make sense to partially support them? Perhaps we should just ban overflows and leave LSP compatibility logic to LSP-related parts of our ecosystem?